Improve lifecycling docs after feedback (#473)
digital-asset/docs.daml.com: docs/2.8.0/docs/daml-finance/concepts/lifecycling.rst

   :alt: The issuer issues the ACME instrument. The investor owns a holding of 1000 ACME shares
         (version 1). The holding references the instrument.

We will now explain each step in the process in detail. The goal is that every holder of a v1

> "cum-dividend" instrument will afterwards have $10 (for each share held) and a v2 "ex-dividend"

Reviewer: ```suggestion
"cum-dividend" instrument will afterwards have $10 (for each unit of instrument held) and a holding on the v2 "ex-dividend"
```

   :alt: The issuer creates a new ACME v2 instrument. They also create a distribution
         event by declaring a dividend on the ACME v1 instrument.

> Note that there is only one instrument (the v1) paying a dividend, so there will be one distribution

Reviewer: The feedback from Shaul is to remind the reader that only one instrument exists, rather than the implication one instrument => one event.

I would suggest something along the lines of

```suggestion
The lifecycle event will now be used to evolve the single instrument contract for the v1 (dividend paying) instrument.
```

wdyt?

Author: Not sure I understand the suggested formulation. In order to remove the implication one instrument => one event, what about something like:
_Note that there is only one instrument (the v1) paying a dividend. The lifecycle event will now be used to evolve the v1 contract into a v2, that does not pay a dividend._

Reviewer: Aha, I think we have a different understanding of the feedback item at hand. 

> In creating-the-event I recommend reminding the reader that there is only one instrument, so there will be one distribution event

I read it as
- remind the reader that there is only one instrument contract that needs to be processed

I believe your interpretation is
- remind the reader that the event will be used to evolve `V1` and not `V1` and `V2`. 

I like yours better :)

To be fair, I don't know why we do not encode the creation of the `V2` instrument directly in the distribution rule, it would make things simpler.

Author: > To be fair, I don't know why we do not encode the creation of the `V2` instrument directly in the distribution rule, it would make things simpler.

Do you want to open a line in the UX sheet for this?

Author: I have reformulated it now.

   :alt: The issuer processes the distribution event through the distribution rule, creating a
         lifecycle effect. The effect references ACME v1 as a target instrument.

> These components are described in more detail in the Components section below. For simplicity, we

Reviewer: add link to section?

have chosen a dividend as an example here, but the same workflow can be used to process more

> advanced instruments involving more complex calculations and where the resulting v2 instrument is

Reviewer: ```suggestion
complex events.
```

I would then move the sentence from above "Note
that the logic to create the new version of an instrument can also be encoded in the lifecycle rule." here.

It doesn't make sense where it currently sits, as we haven't introduced lifecycle rules yet.

Author: If I commit this suggestion, I think it would read 
_, but the same workflow can be used to process more complex events
automatically defined and produced._

Do you mean the following?
_, but the same workflow can be used to process more
complex events and where the resulting v2 instrument is
automatically defined and produced._

Reviewer: I meant to remove the "and where the ..." and replace it with the sentence "Note
that the logic to create the new version of an instrument can also be encoded in the lifecycle rule."

- The payment of the cash dividend amount corresponding to the number of stocks held

Both legs of this settlement are grouped in a
:ref:`Batch <type-daml-finance-interface-settlement-batch-batch-97497>` to provide atomicity. The

> goal of the batch is to take a v1 holding and return a v2 holding + $10 (for each share held). This

Reviewer: ```suggestion
goal of the batch is to exchange a holding on the v1 instrument for a holding on the v2 instrument + $10 (for each share held). This
```

Note that some instruments can require both types of lifecycling. An example of this is a callable
bond that is callable only on some of the coupon dates.

Effects
=======

An :ref:`Effect <type-daml-finance-interface-lifecycle-effect-effect-69649>` describes the asset
movements resulting from a particular event. It specifies these movements per unit of a target
instrument and version. Holdings on this specific instrument version entitle a holder to claim the

> effect, which results in the required asset movements to be instructed. In our dividend example

Reviewer: ```suggestion
effect, which results in the required asset movements to be instructed.

In our dividend example
```

above, this would describe the following asset movement: take a v1 holding and return a v2 holding
+ $10 (per unit held). The actual movement of these assets is described in the next section.

Claim Rule
==========

The :ref:`Claim Rule <type-daml-finance-interface-lifecycle-rule-claim-claim-29284>` is used to

> claim lifecycle effects and instruct settlement thereof. Each effect specifies a target instrument

Reviewer: ```suggestion
claim lifecycle effects and instruct settlement of the resulting asset movements. Each effect specifies a target instrument
```

(and version), and holdings on this instrument (version) are required to claim an effect. In our

> dividend example above, the v1 instrument is the target instrument. This serves as proof of

Reviewer: I would reformulate for clarity (not clear what "This" refers to).
